core/src/impl/Kokkos_FunctorAdapter.hpp

```diff
@@ -177,9 +177,13 @@ struct FunctorValueTraits<FunctorType, ArgTag,
       value_type;
   typedef FunctorType functor_type;
 
-  static_assert(0 == (sizeof(value_type) % sizeof(int)),
-                "Reduction functor's declared value_type requires: 0 == "
-                "sizeof(value_type) % sizeof(int)");
+  // It seems that if you have redction_identities for unsigned char and short
+  // that the below static assert is contradictory.  I am commenting it out
+  // (JSM)
+  //
+  //  static_assert(0 == (sizeof(value_type) % sizeof(int)),
+  //                "Reduction functor's declared value_type requires: 0 == "
+  //                "sizeof(value_type) % sizeof(int)");
```

Comment:
This PR is trying to make this requirement go away right? For now probably we want `static_assert((sizeof(value_type)<sizeof(int)) || (0 == (sizeof(value_type)%sizeof(int)))` because what we did only fixed stuff for smaller 4byte but not for say a 6 byte struct.